components/utils/dataproviders.d.ts

```diff
@@ -2,12 +2,65 @@
 
 declare module JSONForms{
     export interface IDataProvider extends IService {
+
+        /**
+         * Initially fetches the data instance
+         *
+         * @return a promise returning the initially fetched data
+         */
         fetchData(): ng.IPromise<any>
-        fetchPage(page:number, size:number): ng.IPromise<any>
-        setPageSize(size:number)
+
+
+        /**
+         * Returns the instance.
+         *
+         * @return {any} the instance
+         */
         getData(): any
-        getPageSize(): number
-        getPage(): number
+
```

Comment:
I think we could, of course, but I think paging/filtering should stay optional nevertheless. Anyways, I think we should discuss in a separate issue, what do you think?